src/rulesets/Base/fastmath_able.jl

```diff
@@ -5,8 +5,6 @@ let
     # e.g. do not add `foo` unless `Base.FastMath.foo_fast` exists.
     fastable_ast = quote
         #  Trig-Basics
-        @scalar_rule cos(x) -(sin(x))
```

Comment:
Why were these rules moved out of fastmath-able?

Reply:
The reason I had in mind was, I saw the definition of `@scalar_rule` which says:
```
A convenience macro that generates simple scalar forward or reverse rules using the provided partial derivatives. Specifically, generates the corresponding methods for frule and rrule:
```

But since we're explicitly writing down the `frule` and `rrule` for both `sin` and `cos`, so they would probably not be needed no more. I'm not sure as to where I might be wrong in thinking like that.

Reply:
we want the new rules that replace this to also be in the `fastmath-able` file/quote.
Because then it will also generate matching ruls for `Base.FastMath.sin_fast` and `Base.FastMath.cos_fast`, which will internally use `Base.FastMath.sincos_fast`.
Basically the way that file works is it makes a copy of the code, and rewrites it at the AST level to replace all suitable operations (both in body and in signature) with their FastMath equivelents, and then evaluated both the original AST and the new fastmath varient, thus defining rules for both.